source/MRViewer/MRRibbonMenu.cpp

```diff
@@ -1710,102 +1710,326 @@ void RibbonMenu::setupShortcuts_()
+    auto shortcutListIt = shortcutList.begin();
+
+    auto addReadOnlyLine = [scaling, &style] ( const std::string& line )
+    {
+        ImGui::PushStyleColor( ImGuiCol_Text, Color::gray().getUInt32() );           
+
+        const auto textWidth = ImGui::CalcTextSize( line.c_str() ).x;
+        // read only so const_sast should be ok
+        auto itemWidth = std::max( textWidth + 2 * style.FramePadding.x, 30.0f * scaling );
+        ImGui::PushItemWidth( itemWidth );
+
+        auto framePaddingX = std::max( style.FramePadding.x, ( itemWidth - textWidth ) / 2.0f );
+        ImGui::PushStyleVar( ImGuiStyleVar_FramePadding, { framePaddingX, cButtonPadding * scaling } );
+        ImGui::InputText( ( "##" + line ).c_str(), const_cast< std::string& >( line ), ImGuiInputTextFlags_ReadOnly | ImGuiInputTextFlags_AutoSelectAll );
+        ImGui::PopItemWidth();        
+        ImGui::PopStyleColor();
+        ImGui::PopStyleVar();
+    };
+
+    const auto columnWidth = ( windowWidth - 2 * style.WindowPadding.x ) / 2;
+    
+    ImGui::SetCursorPosY( ImGui::GetCursorPosY() + cDefaultItemSpacing * scaling );
+
+    if ( ImGui::BeginTable( "HotKeysTable", 2, ImGuiTableFlags_SizingStretchSame ) )
+    {
+        const int categoryCount = static_cast< int >( ShortcutManager::ShortcutCategory::Count );
+        ImGui::PushStyleVar( ImGuiStyleVar_FramePadding, { cButtonPadding * scaling, cButtonPadding * scaling } );
+
+        ImGui::TableNextColumn();
+        for ( int i = 0; i < categoryCount; ++i )
+        {
+            if ( i == categoryCount / 2  )
+            {
+                ImGui::TableNextColumn();
+                ImGui::Indent();
+            }
+
+            ImGui::PushFont( fontManager_.getFontByType( MR::RibbonFontManager::FontType::BigSemiBold ) );
+            ImGui::Separator( scaling, categoryNames[i].c_str() );
+            ImGui::PopFont();
+
+            auto [key, category, text] = *shortcutListIt;
+
+            while ( shortcutListIt != std::end( shortcutList ) && static_cast<int>( category ) == i )
+            {
+                auto transparentColor = ImGui::GetStyleColorVec4( ImGuiCol_Text );
+                transparentColor.w *= 0.5f;
+                ImGui::PushStyleColor( ImGuiCol_Text, transparentColor );
+                ImGui::Text( "%s", text.c_str());
+                ImGui::PopStyleColor();
+
+                float paddings = 2 * style.FramePadding.x + 3 * style.ItemInnerSpacing.x;
+                if ( i >= categoryCount / 2 )
+                    paddings += style.IndentSpacing;
+
+                if ( key.mod & GLFW_MOD_CONTROL )
+                {
+                    paddings += 2 * style.FramePadding.x + 2 * style.ItemInnerSpacing.x;
+                    text += "Ctrl+";
+                }
+                if ( key.mod & GLFW_MOD_ALT )
+                {
+                    paddings += 2 * style.FramePadding.x + 2 * style.ItemInnerSpacing.x;
+                    text += "Alt+";
+                }
```

Comment:
can we make such logic near (separate to smaller public function e.g. get modifier strig, get key string)
```c++
    // make string form key and returns it
    MRVIEWER_API static std::string ShortcutManager::getKeyString( const ShortcutKey& key );